tests/unit/mathematics/RandomMixin_unittest.cc

```diff
@@ -0,0 +1,176 @@
+#include <shogun/lib/common.h>
+#include <shogun/lib/config.h>
+#include <shogun/base/SGObject.h>
+#include <shogun/mathematics/RandomMixin.h>
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <array>
+
+using namespace shogun;
+
+class MockObject : public CSGObject
```

Comment:
there's already a mockobject within unit tests fw... would be great if we would have one if possible in the repo :)